packages/webpack-extensive-lodash-replacement-plugin/index.js

```diff
@@ -443,15 +478,22 @@ class ExtensiveLodashReplacementPlugin {
 	}
 
 	apply( compiler ) {
+		compiler.hooks.thisCompilation.tap( 'LodashReplacementPlugin', compilation => {
+			this.compilation = compilation;
+		} );
+
 		compiler.hooks.normalModuleFactory.tap( 'LodashReplacementPlugin', nmf => {
-			nmf.hooks.beforeResolve.tap( 'LodashReplacementPlugin', result => {
+			this.moduleResolver = this.moduleResolver || nmf.getResolver( 'normal' );
+			this.initBaseLodashData();
+
+			nmf.hooks.beforeResolve.tapPromise( 'LodashReplacementPlugin', async result => {
 				if ( ! result ) return;
-				result.request = this.getModifiedRequest( result );
+				result.request = await this.getModifiedRequest( result );
```

Comment:
Ideally, if `getModifiedRequest` is not going to modify the request, the `result` shouldn't be touched at all.

I would personally put everything into a `modifyRequest` method:
```js
async modifyRequest( result ) {
  if ( ! result ) {
    return;
  }
  if ( ! result.contextInfo ) {
    return;
  }
  if ( await canBeReplaced( ... ) ) {
    request.result = ..;
    return;
  }
}

tapPromise( result => this.modifyRequest.bind( this ) );
```

Reply:
I named it `modifyResult`, but I followed your pattern 👍